instance/tasks.py

```diff
@@ -339,3 +345,103 @@ def cleanup_old_betatest_users():
         users_to_delete = queryset.filter(is_active=False, profile__modified__lte=delete_cutoff)
         logger.info('Deleting %s users.', users_to_delete.count())
         users_to_delete.delete()
+
+
+class KillZombiesRunner:
+    """
+    Helper class to run `kill_zombies_periodically`
+    """
+
+    def __init__(self):
+        self.enabled: bool = bool(os.environ.get("KILL_ZOMBIES_ENABLED", False))
+        self.region: str = os.environ.get("KILL_ZOMBIES_REGION", "gra")
+        self.recipient: str = os.environ.get(
+            "KILL_ZOMBIES_WARNING_RECIPIENT", "ops@opencraft.com"
+        )
+        self.threshold: int = int(os.environ.get("KILL_ZOMBIES_THRESHOLD", 10))
+
+    def send_email(self, subject: str, body: str) -> int:
+        """
+        Utility method for sending emails
+        """
+        email = EmailMessage(
+            subject, body, settings.DEFAULT_FROM_EMAIL, [self.recipient]
+        )
+        return email.send()
+
+    def trigger_warning(self):
+        """
+        Warns when more than `threshold` VMs will be terminated
+        """
+        subject = "{} Zombie instances over current threshold ({})".format(
+            datetime.now().strftime("%Y-%m-d"), self.threshold
+        )
+        body = (
+            "The number of instances in region {} to be terminated"
+            "is over KILL_ZOMBIES_WARNING_THRESHOLD ({}).\n"
+        ).format(self.region, self.threshold)
+        logging.info(
+            "%s\nSending an email notification to: %s",
+            body, self.recipient
+        )
+        self.send_email(subject, body)
+
+    def handle_dry_run_output(self, stdout: str) -> int:
+        """
+        If there are servers to terminate, a `kill_zombies` dry run will warn:
+        "Would have terminated {} zombies" to stdout.
+        """
+        if "No servers found in region" in stdout:
+            return 0
+        try:
+            return int(stdout.split("Would have terminated")[1].split()[0])
```

Comment:
@spokerman12, is there a better/cleaner way to get the number from the output? I can think of using a regular expression, but you know what they say
> Some people, when confronted with a problem, think "I know, I'll use regular expressions." Now they have two problems.

😛 
In this case, it looks like using a regular expression might make this cleaner. The saying warns against overusing/using regular expressions everywhere without assessing if there are better solutions :)

Reply:
Oh of course, I love regex and I considered it but because we need to find `Would have terminated` and then the number, most reasonable implementations of this line need at least two calls, that is without using a more complex regex.
I tried once again and it seems a bit unnecessary for such a predictable log output.
Thanks for the suggestion, though!